src/main/scala/akka/persistence/cassandra/EventsByPersistenceIdIterator.scala

```diff
@@ -0,0 +1,84 @@
+package akka.persistence.cassandra
+
+import java.lang.{Long => JLong}
+import java.nio.ByteBuffer
+
+import scala.collection.JavaConverters._
+
+import akka.persistence.PersistentRepr
+import akka.persistence.PersistentRepr._
+import akka.persistence.cassandra.query.journal.{CassandraReadStatements, CassandraReadJournalConfig}
+import akka.serialization.Serialization
+import com.datastax.driver.core.utils.Bytes
+import com.datastax.driver.core.{ResultSet, Row, Session}
+
+final class EventsByPersistenceIdIterator(
```

Comment:
The iterators extracted to separate files for reuse.

This is not ideal and should be asynchronous, especially in query publishers. But that is easily amended as seen is query actor publisher where it is abstracted by use of futures.